File: app/internal/publisher/template/dotnet/RenewalDialog.cs
using System;
using System.IO;
using System.Text.Json;
using System.Threading.Tasks;

namespace WebsmithSDK;

public class RenewalDialog
{
    private readonly ApiClient _client;
    private readonly CacheManager _cache;
    private readonly string _supportEmail;

    public class RenewalDialogResult
    {
        public bool Renewed { get; set; }
        public bool Cancelled { get; set; }
        public string? LicenseKey { get; set; }
        public string? Expiry { get; set; }
        public string? Error { get; set; }
    }

    public RenewalDialog(ApiClient? client)
    {
        _client = client ?? new ApiClient();
        _cache = _client.GetCache();
        _supportEmail = LoadSupportEmail();
    }

    private static string LoadSupportEmail()
    {
        try
        {
            var path = "config/api-config.json";
            if (File.Exists(path))
            {
                var doc = JsonDocument.Parse(File.ReadAllText(path));
                if (doc.RootElement.TryGetProperty("branding", out var branding)
                    && branding.TryGetProperty("support_email", out var email))
                {
                    return email.GetString() ?? "support@websmithdigital.com";
                }
            }
        }
        catch { }
        return "support@websmithdigital.com";
    }

    public RenewalDialogResult Show(string? licenseKey)
    {
        if (string.IsNullOrEmpty(licenseKey))
        {
            Console.WriteLine("\n=== License Renewal ===\n");
            Console.Write("Enter License Key to renew: ");
            licenseKey = Console.ReadLine()?.Trim();
            if (string.IsNullOrEmpty(licenseKey))
            {
                Console.WriteLine("License key is required.");
                return new RenewalDialogResult { Cancelled = true, Error = "No license key entered" };
            }
        }

        Console.Write("\nHow many days to renew? (leave empty for default): ");
        var input = Console.ReadLine()?.Trim() ?? "";
        int? extraDays = null;
        if (!string.IsNullOrEmpty(input) && int.TryParse(input, out var days))
        {
            extraDays = days;
        }

        Console.WriteLine($"\nProcessing renewal for license key: {licenseKey}");
        if (extraDays.HasValue)
            Console.WriteLine($"Extra days: {extraDays.Value}");

        try
        {
            var result = _client.RenewLicense(licenseKey, extraDays).Result;
            if (result.RootElement.TryGetProperty("success", out var s) && s.GetBoolean())
            {
                JsonElement data;
                if (result.RootElement.TryGetProperty("data", out var d))
                    data = d;
                else
                    data = result.RootElement;

                var expiry = data.TryGetProperty("expiry_date", out var ed)
                    ? ed.GetString()
                    : data.TryGetProperty("expires_at", out var ea) ? ea.GetString() : "N/A";

                Console.WriteLine($"\nLicense renewed successfully!");
                Console.WriteLine($"New expiry: {expiry}");
                _cache.InvalidateLicenseStatus();
                return new RenewalDialogResult { Renewed = true, LicenseKey = licenseKey, Expiry = expiry };
            }
            else
            {
                var errMsg = result.RootElement.TryGetProperty("message", out var msg)
                    ? msg.GetString()
                    : result.RootElement.TryGetProperty("error", out var e)
                        ? e.GetString() : "Renewal failed";
                Console.WriteLine($"\nRenewal failed: {errMsg}");
                return new RenewalDialogResult { Error = errMsg };
            }
        }
        catch (Exception e)
        {
            Console.WriteLine($"\nError: {e.Message}");
            return new RenewalDialogResult { Cancelled = true, Error = e.Message };
        }
    }
}

File: app/internal/publisher/template/java/RenewalDialog.java
package com.websmith.sdk;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.io.File;
import java.io.FileReader;
import java.util.Scanner;

public class RenewalDialog {

    private ApiClient client;
    private CacheManager cache;
    private String productName;
    private String supportEmail;

    public static class RenewalDialogResult {
        public boolean renewed;
        public boolean cancelled;
        public String licenseKey;
        public String expiry;
        public String error;

        public RenewalDialogResult(boolean renewed, boolean cancelled, String licenseKey,
                                    String expiry, String error) {
            this.renewed = renewed;
            this.cancelled = cancelled;
            this.licenseKey = licenseKey;
            this.expiry = expiry;
            this.error = error;
        }
    }

    public RenewalDialog(ApiClient client, String productName) {
        this.client = client;
        this.cache = client != null ? client.getCache() : new CacheManager(loadConfig());
        this.productName = productName != null ? productName : "";
        JsonObject config = loadConfig();
        if (config.has("branding")) {
            JsonObject branding = config.getAsJsonObject("branding");
            this.supportEmail = branding.has("support_email") ? branding.get("support_email").getAsString() : "support@websmithdigital.com";
        } else {
            this.supportEmail = "support@websmithdigital.com";
        }
    }

    public RenewalDialog(ApiClient client) {
        this(client, null);
    }

    private static JsonObject loadConfig() {
        String[] paths = {"config/api-config.json", "config/api-config.json"};
        for (String path : paths) {
            try {
                File f = new File(path);
                if (f.exists()) {
                    try (FileReader reader = new FileReader(f)) {
                        return JsonParser.parseReader(reader).getAsJsonObject();
                    }
                }
            } catch (Exception ignored) {}
        }
        return new JsonObject();
    }

    public RenewalDialogResult show(String licenseKey) {
        Scanner scanner = new Scanner(System.in);

        if (licenseKey == null || licenseKey.isEmpty()) {
            System.out.println("\n=== License Renewal ===\n");
            System.out.print("Enter License Key to renew: ");
            licenseKey = scanner.nextLine().trim();
            if (licenseKey.isEmpty()) {
                System.out.println("License key is required.");
                return new RenewalDialogResult(false, true, null, null, "No license key entered");
            }
        }

        System.out.print("\nHow many days to renew? (leave empty for default): ");
        String input = scanner.nextLine().trim();
        Integer extraDays = null;
        if (!input.isEmpty()) {
            try {
                extraDays = Integer.parseInt(input);
            } catch (NumberFormatException e) {
                System.out.println("Invalid number. Using default duration.");
            }
        }

        System.out.println("\nProcessing renewal for license key: " + licenseKey);
        if (extraDays != null) {
            System.out.println("Extra days: " + extraDays);
        }

        try {
            JsonObject result = client.renewLicense(licenseKey, extraDays);
            if (result.has("success") && result.get("success").getAsBoolean()) {
                JsonObject data = result.has("data") ? result.getAsJsonObject("data") : result;
                String expiry = data.has("expiry_date") ? data.get("expiry_date").getAsString()
                    : data.has("expires_at") ? data.get("expires_at").getAsString() : "N/A";
                System.out.println("\nLicense renewed successfully!");
                System.out.println("New expiry: " + expiry);
                cache.invalidateLicenseStatus();
                return new RenewalDialogResult(true, false, licenseKey, expiry, null);
            } else {
                String errMsg = result.has("message") ? result.get("message").getAsString()
                    : result.has("error") ? result.get("error").getAsString()
                    : "Renewal failed";
                System.out.println("\nRenewal failed: " + errMsg);
                return new RenewalDialogResult(false, false, null, null, errMsg);
            }
        } catch (Exception e) {
            System.out.println("\nError: " + e.getMessage());
            return new RenewalDialogResult(false, true, null, null, e.getMessage());
        }
    }
}
